### PR TITLE
Dashboard

### PR DESCRIPTION
**Pull request by upexcel**

### CODE REVIEW

File: both/collections/csvdata.collection.ts [hunk 1]
```diff
@@ -1,159 +1,317 @@
...
+                }
+
+
+            }
+        }
+        return true;
+    }, // Meteor method addcategory will assign category to our document which we choose in csvjson component
+
+    'addCategory' (id, category) {
+        check(id, String);
```
Comment (reactexcel) [inline both/collections/csvdata.collection.ts]: can we check here only admin can add category

File: both/collections/csvdata.collection.ts [hunk 2]
```diff
@@ -1,159 +1,317 @@
...
+            "_id": id
+        }, {
+            $set: {
+                "Assigned_category": category,
+                "is_processed": 1
+            }
+        });
+    },
+    'changeCategory' (id, category) {
+        check(id, String);
```
Comment (reactexcel) [inline both/collections/csvdata.collection.ts]: can we check here only admin can add category

File: both/collections/csvdata.collection.ts [hunk 3]
```diff
@@ -1,159 +1,317 @@
...
+        Csvdata.update({
+            "_id": id
+        }, {
+            $set: {
+                "Assigned_category": category
+            }
+        });
+    },
+    'addInvoice' (id, invoice_no, description, linkarray) {
+        check(id, String);
```
Comment (reactexcel) [inline both/collections/csvdata.collection.ts]: can we add check here only admin and accounts can do this

File: both/collections/csvdata.collection.ts [hunk 4]
```diff
@@ -1,159 +1,317 @@
...
+        }, {
+            $set: {
+                "invoice_no": invoice_no,
+                "invoice_description": description,
+                "linktodrive": linkarray
+            }
+        });
+    },
+    'deleteInvoice' (id) {
+        check(id, String);
```
Comment (reactexcel) [inline both/collections/csvdata.collection.ts]: can we add check here only admin and accounts can do this

File: both/collections/csvdata.collection.ts [hunk 5]
```diff
@@ -1,159 +1,317 @@
...
+        }, {
+            $set: {
+                "invoice_no": "not_assigned",
+                "invoice_description": "invoice_description",
+                "linktodrive": "notassigned"
+            }
+        });
+    },
+    'addUser' (adduserinfo) {
+        check(adduserinfo.username, String);
```
Comment (reactexcel) [inline both/collections/csvdata.collection.ts]: add check only admin can do this

File: both/collections/csvdata.collection.ts [hunk 6]
```diff
@@ -1,159 +1,317 @@
...
+                let userid = Accounts.createUser(adduserinfo);
+                Roles.addUsersToRoles(userid, [adduserinfo.profile.role]);
+            } else {
+                throw new Meteor.Error(403, "Access denied");
+            }
+        }
+
+    },
+    'removeUser' (user) {
+        if (Roles.userIsInRole(Meteor.userId(), 'admin')) {
```
Comment (reactexcel) [inline both/collections/csvdata.collection.ts]: only admin

File: both/collections/csvdata.collection.ts [hunk 7]
```diff
@@ -1,159 +1,317 @@
...
+                Accounts.setPassword(userId, newPassword);
+            } else {
+                throw new Meteor.Error(403, "Access denied");
+            }
         }
-        else{
-             throw new Meteor.Error(403, "Access denied");
+    },
+    'assignTransDocToUser' (docid, userid, username) {
+        if (Meteor.isServer) {
```
Comment (reactexcel) [inline both/collections/csvdata.collection.ts]: only admin and accounts

File: client/modules/core/addproduct/addproduct.component.ts [hunk 8]
```diff
@@ -40,33 +37,25 @@ import template from './addproduct.html';
 export class CsvAddProductComponent implements OnInit, OnDestroy {
```
Comment (reactexcel) [inline client/modules/core/addproduct/addproduct.component.ts]: why is this called addProduct?

File: client/modules/core/dashboard/dashboard.ts [hunk 9]
```diff
@@ -0,0 +1,148 @@
...
+})
+
+export class DashboardComponent implements OnInit, OnDestroy {
+
+    current_year_header: any;
+    current_year: number;
+    date: any;
+    csvdata: any;
+    csvSub: Subscription;
+    chartData: any = []
```
Comment (reactexcel) [inline client/modules/core/dashboard/dashboard.ts]: code formatting

File: client/modules/core/dashboard/dashboard.ts [hunk 10]
```diff
@@ -0,0 +1,148 @@
...
+                        'label': 'Credit'
+                    }];
+                }
+            })
+        } //**** ngInit ends here ****
+
+        charData = [{
+        data: [0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0],
+        label: 'Deposit'
+    }, {
```
Comment (reactexcel) [inline client/modules/core/dashboard/dashboard.ts]: formatting

File: client/modules/core/dashboard/dashboard.ts [hunk 11]
```diff
@@ -0,0 +1,148 @@
...
+    };
+    public barChartLabels: string[] = ['January', 'February', 'March', 'April', 'May', 'June', 'July', 'August', 'September', 'October', 'November', 'December'];
+    public barChartType: string = 'bar';
+    public barChartLegend: boolean = true;
+
+
+
+    // events
+    public chartClicked(e: any): void {
+        console.log(e);
```
Comment (reactexcel) [inline client/modules/core/dashboard/dashboard.ts]: remove

File: client/modules/core/dashboard/dashboard.ts [hunk 12]
```diff
@@ -0,0 +1,148 @@
...
+            this.csvSub.unsubscribe();
+        }
+        var options = {
+            year: newdate
+        };
+        this.csvdata = Graphdata.find().zone();
+        this.csvSub = MeteorObservable.subscribe('csvdata_month', options).subscribe(() => {
+
+        });
+        this.csvdata.subscribe((data) => {
```
Comment (reactexcel) [inline client/modules/core/dashboard/dashboard.ts]: move this code to a private function. dont write same code multiple times

File: client/modules/core/dashboard/dashboard.ts [hunk 13]
```diff
@@ -0,0 +1,148 @@
...
+} from '../../../../both/collections/csvdata.collection';
+
+@Component({
+    selector: 'dashboard',
+    template
+})
+
+export class DashboardComponent implements OnInit, OnDestroy {
+
+    current_year_header: any;
```
Comment (reactexcel) [inline client/modules/core/dashboard/dashboard.ts]: if admin user logins, he should automatically go to dashboard.

if accounts or guest user logins he should go to transactions page

File: both/collections/csvdata.collection.ts (hunk 3 above)
Comment (upexcel) [inline both/collections/csvdata.collection.ts]: done

File: client/modules/core/addproduct/addproduct.component.ts (hunk 8 above)
Comment (upexcel) [inline client/modules/core/addproduct/addproduct.component.ts]: now changed to addcategory

File: both/collections/csvdata.collection.ts (hunk 2 above)
Comment (upexcel) [inline both/collections/csvdata.collection.ts]: change category is use if we want to change a category that is assigned to any transaction note. now only admin can change this category

File: both/collections/csvdata.collection.ts (hunk 5 above)
Comment (upexcel) [inline both/collections/csvdata.collection.ts]: done

File: client/modules/core/dashboard/dashboard.ts (hunk 13 above)
Comment (upexcel) [inline client/modules/core/dashboard/dashboard.ts]: done

File: client/modules/core/dashboard/dashboard.ts (hunk 10 above)
Comment (upexcel) [inline client/modules/core/dashboard/dashboard.ts]: i have used jsbeautifier for formatting.

File: client/modules/core/dashboard/dashboard.ts (hunk 11 above)
Comment (upexcel) [inline client/modules/core/dashboard/dashboard.ts]: removed

File: client/modules/core/dashboard/dashboard.ts (hunk 12 above)
Comment (upexcel) [inline client/modules/core/dashboard/dashboard.ts]: done

**Pull request merged**